src/validation.cpp

```diff
@@ -3864,10 +3957,21 @@ bool RewindBlockIndex(const CChainParams& params)
         PruneBlockIndexCandidates();
 
         CheckBlockIndex(params.GetConsensus());
+    }
+
+    return true;
+}
+
+bool RewindBlockIndex(const CChainParams& params) {
```

Comment:
In commit "Move a bunch of chainstate-manipulation functions into CChainState"

Two RewindBlockIndex functions with the same name & params but slightly different behavior seems not great.

Maybe should have RewindBlockIndexDB / CChainState::RewindBlockIndex, analogous to LoadBlockIndexDB / CChainState::LoadBlockIndex.

Reply:
They dont (really) have different behavior, just the public wrapper version does a DB flush, but that isnt strictly required (I believe). But, really, CChainState/validation should have no concept of the DB backend in use/the disk, hence the reason for the flush outside, its just a thin wrapper functions to expose it publicly.

Reply:
I'd be fine with a flush in init, but it seems weird given its kinda a validation-disk-storage-layer thing, and I hate adding more low-level logic in init.